docs/deployment/resource-providers/yarn.md

```diff
@@ -0,0 +1,238 @@
+echo "stop" | ./bin/yarn-session.sh -id application_XXXXX_XXX
+{% endhighlight %}
+
+Congratulations! You have successfully run a Flink application by deploying Flink on YARN.
+
+{% top %}
+
+## Deployment Modes Supported by Flink on YARN
+
+For production use, we recommend deploying Flink Applications in the [Per-job or Application Mode]({% link deployment/index.md %}#deployment-modes), as these modes provide a better isolation for the Applications.
+
+### Application Mode
+
+Application Mode will launch a Flink cluster on YARN, where the main() method of the application jar gets executed on the JobManager in YARN.
+The cluster will shut down as soon as the application has finished. You can manually stop the cluster using `yarn application -kill <ApplicationId>` or by cancelling the Flink job.
+
+{% highlight bash %}
+./bin/flink run-application -t yarn-application ./examples/streaming/TopSpeedWindowing.jar
+{% endhighlight %}
+
+
+Once an Application Mode cluster is deployed, you can interact with it for operations like cancelling or taking a savepoint.
+
+{% highlight bash %}
+# List running job on the cluster
+./bin/flink list -t yarn-application -Dyarn.application.id=application_XXXX_YY
+# Cancel running job
+./bin/flink cancel -t yarn-application -Dyarn.application.id=application_XXXX_YY <jobId>
+{% endhighlight %}
+
+Note that cancelling your job on an Application Cluster will stop the cluster.
+
+To unlock the full potential of the application mode, consider using it with the `yarn.provided.lib.dirs` configuration option
+and pre-upload your application jar to a location accessible by all nodes in your cluster. In this case, the 
+command could look like: 
+
+{% highlight bash %}
+./bin/flink run-application -t yarn-application \
+	-Dyarn.provided.lib.dirs="hdfs://myhdfs/my-remote-flink-dist-dir" \
+	hdfs://myhdfs/jars/my-application.jar
+{% endhighlight %}
+
+The above will allow the job submission to be extra lightweight as the needed Flink jars and the application jar
+are  going to be picked up by the specified remote locations rather than be shipped to the cluster by the 
+client.
+
+### Per-Job Cluster Mode
+
+The Per-job Cluster mode will launch a Flink cluster on YARN, then run the provided application jar locally and finally submit the JobGraph to the JobManager on YARN. If you pass the `--detached` argument, the client will stop once the submission is accepted.
+
+The YARN cluster will stop once the job has stopped.
+
+{% highlight bash %}
+./bin/flink run -t yarn-per-job --detached ./examples/streaming/TopSpeedWindowing.jar
+{% endhighlight %}
+
+Once a Per-Job Cluster is deployed, you can interact with it for operations like cancelling or taking a savepoint.
+
+{% highlight bash %}
+# List running job on the cluster
+./bin/flink list -t yarn-per-job -Dyarn.application.id=application_XXXX_YY
+# Cancel running job
+./bin/flink cancel -t yarn-per-job -Dyarn.application.id=application_XXXX_YY <jobId>
```

Comment:
Same here. It seems a bit odd that one needs to specify `-t yarn-per-job` instead of only `-t yarn` or so.